cli/src/semgrep/git.py

```diff
@@ -333,6 +289,7 @@ def baseline_context(self) -> Iterator[None]:
             # Note that we have no good way of detecting this issue without inspecting the checkout output
             # message, which means we are fragile with respect to git version here.
             logger.debug("Running git reset to return original context")
+            # nosem: use-git-check-output-helper
```

Comment:
why?

Reply:
We use the returncode here for some logic :D